atlp-rwanda/ecommerce-app-champions-bn
# #55password change prompt

# what does this PR do
This pull request adds a feature that prompts users to update their passwords when they expire.

# Description of Task to be completed?
As a user , I should be prompted to update my password
Such that the information about the last time password reset be recorded

# How should this be manually tested
To test this feature manually, the server should be running and a cron job will execute periodically to check if any passwords have expired. If there are any expired passwords, the user with the expired passwords will be sent the email requesting them to change their passwords, and the user statuses will be changed to "NeedToUpdatePassword". To confirm that the feature is working correctly, check the content of the Users table in the database.

. clone this repository
. cd into into
. checkout into ft-password-change-prompt
.add these variables and their values in your .env file  `CRON_SCHEDULE` and  `PASSWORD_EXPIRY`
. run migrations and seeders by `npm run migrate:up` and `npm run migrate:db:seedss respectively
. then start the server by `npm run dev`
. check if expired password are marked and those users the email is sent to them

# What are the relevant pivotal tracker user story
#55
